MAP_POPULATE is not defined in OS X

clickhouse fails to compile on os x:


In file included from 

.../build/contrib/build/src/clickhouse/dbms/src/Common/PODArray.h:15:
.../build/contrib/build/src/clickhouse/dbms/src/Common/Allocator.h:182:32: error: use of undeclared identifier 'MAP_POPULATE'
            | (mmap_populate ? MAP_POPULATE : 0);
                               ^

--- a/dbms/src/Common/Allocator.h
+++ b/dbms/src/Common/Allocator.h
@@ -178,8 +178,13 @@ protected:
     // hash tables, it makes sense to pre-fault the pages by passing
     // MAP_POPULATE to mmap(). This takes some time, but should be faster
     // overall than having a hot loop interrupted by page faults.
+    // It is only supported on Linux.
+#if defined(__linux__)
     static constexpr int mmap_flags = MAP_PRIVATE | MAP_ANONYMOUS
             | (mmap_populate ? MAP_POPULATE : 0);
+#else
+    static constexpr int mmap_flags = MAP_PRIVATE | MAP_ANONYMOUS;
+#endif
 
 private:
     void * allocNoTrack(size_t size, size_t alignment)

